File: app/src/main/java/com/nexaas/pkceclienttest/interfaces/ProfileService.java
package com.nexaas.pkceclienttest.interfaces;

import com.nexaas.pkceclienttest.models.UserProfile;

import retrofit2.Call;
import retrofit2.http.GET;

/**
 * Created by alexandre on 24/08/2018.
 * Copyright© Nexaas™. All rights reserved.
 */

public interface ProfileService {

    @GET ("api/v1/profile")
    Call<UserProfile> getProfile ();

}

File: nexaasidauth/src/main/java/com/nexaas/nexaasid/auth/api/services/ProfileService.kt
package com.nexaas.nexaasid.auth.api.services

import com.nexaas.nexaasid.auth.model.Contacts
import com.nexaas.nexaasid.auth.model.Emails
import com.nexaas.nexaasid.auth.model.PersonalInfo
import com.nexaas.nexaasid.auth.model.ProfessionalInfo
import retrofit2.Call
import retrofit2.http.GET

interface ProfileService {

    @get:GET("api/v1/profile")
    val personalInfo: Call<PersonalInfo>

    @get:GET("api/v1/profile/professional_info")
    val professionalInfo: Call<ProfessionalInfo>

    @get:GET("api/v1/profile/contacts")
    val contacts: Call<Contacts>

    @get:GET("api/v1/profile/emails")
    val emails: Call<Emails>
}